profiles/serializers.py

```diff
@@ -21,11 +21,19 @@ class ProfileSerializer(serializers.ModelSerializer):
     activity = ActivitySerializer(many=True, read_only=True)
     category = CategorySerializer(many=True, read_only=True)
     person = serializers.PrimaryKeyRelatedField(queryset=CustomUser.objects.all())
+    is_saved = serializers.SerializerMethodField()
 
     class Meta:
         model = Profile
         fields = "__all__"
 
+    def get_is_saved(self, obj):
+        user = self.context["request"].user
+        if user.is_authenticated:
+            saved_companies = obj.saved_list.filter(user=user)
+            return saved_companies.exists() 
+        return False
```

Comment:
In case someone will decide to use this serialize in List endpoint, we'll have a trouble. 

For _each_ of the profile, this code (containing call to DB) will be executed, so there will be N+1 problem.

If we're sure that the serializer will be used only for details endpoint (and in this case adding `Details` word into the name would be helpful), we can keep it as is.

If not (let's say, we know for sure that it will be used for List endpoint), we can do the next trick:

1. Retrieve PKs of all comanies, saved by current user.
2. Put this data as a list (or, better — frozenset) into serializer context (by overriding `get_serializer_context()`)
3. This method will contain a single string of code, something like `return obj.pk in self.context['saved_companies_pks']`
4. .. and also, you can combine these approaches — check if there is `saved_companies_pks` in context, and do the next steps based on the result.